bindings/python/CMakeLists.txt

```diff
@@ -6,6 +6,7 @@ include(${JRL_CMAKE_MODULES}/python.cmake)
 include(${JRL_CMAKE_MODULES}/python-helpers.cmake)
 
 findpython(REQUIRED Development.Module)
+find_package(Python 3.7 COMPONENTS ${PYTHON_COMPONENTS})
```

Comment:
Hi, yeah we can see this inconsistent behavior [here](https://github.com/Simple-Robotics/proxsuite/actions/runs/10298582919/job/28504030168#step:9:61), while it was not the case [before](https://github.com/Simple-Robotics/proxsuite/actions/runs/10298582919/job/28504030168#step:9:61). I think it might be related to the change you introduce here.

Reply:
Is the findpython from our modules a fix/wrappper for find_package(Python)?

I think I just followed nanobind's setup tutorial which used find_package

Reply:
it is from the jrl-cmakemodules: https://github.com/jrl-umi3218/jrl-cmakemodules/blob/master/python.cmake#L16

I think we should just remove the line you have added

Reply:
It seems nanobind's CMake file does part of the work for us:
https://github.com/wjakob/nanobind/blob/b1531b9397c448c6b784520a4f052608f28a5e8d/CMakeLists.txt#L147

Also, nanobind links to modern CMake targets prefixed by `Python::` (not `Python3::`) and not to like variables like `${PythonLibs}`.